packages/sdk/src/HeadlessUniversalCheckout/Managers/RedirectManager.ts

```diff
@@ -0,0 +1,208 @@
+import type { PrimerError } from '@primer-io/react-native';
+import {
+  NativeEventEmitter,
+  NativeModules,
+  EmitterSubscription,
+} from 'react-native';
+import { PrimerInitializationData } from 'src/models/PrimerInitializationData';
+
+const { RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager } =
+  NativeModules;
+
+const eventEmitter = new NativeEventEmitter(
+  RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager
+);
+
+type EventType = 'onRetrieved' | 'onRetrieving' | 'onInvalid' | 'onValid' | 'onError' | 'onValidating';
+
+const eventTypes: EventType[] = [
+  'onRetrieved',
+  'onRetrieving',
+  'onInvalid',
+  'onError',
+  'onValid',
+  'onValidating'
+];
+
+export interface RedirectManagerProps {
+  paymentMethodType: string;
+  onRetrieved?: (metadata: any) => void;
+  onRetrieving?: () => void;
+  onError?: (errors: PrimerError[] | undefined) => void;
+  onInvalid?: (isValid: any) => void;
+  onValid?: () => void;
+  onValidating?: () => void;
+}
+
+class PrimerHeadlessUniversalCheckoutComponentWithRedirectManager {
+  options?: RedirectManagerProps;
+
+  ///////////////////////////////////////////
+  // Init
+  ///////////////////////////////////////////
+  constructor() { }
+
+  ///////////////////////////////////////////
+  // API
+  ///////////////////////////////////////////
+
+  async configure(
+    options: RedirectManagerProps
+  ): Promise<{ initializationData: PrimerInitializationData } | void> {
```

Comment:
@NQuinn27, @semirp, is PrimerInitializationData being used as a return type expected here?

Reply:
I'm asking because it appears that PrimerInitializationData has a single implementation, RetailOutletsList, and RetailOutletsList is not used with Adyen iDeal.

Reply:
not really, we don't use that here